src/EventGenerator/EventGenerator.php

```diff
@@ -145,4 +154,42 @@ public function generateEvent(EntityInterface $entity, UserInterface $user, arra
     return json_encode($event);
   }
 
+  /**
+   * Method to check if an entity is a new revision.
+   *
+   * @param \Drupal\Core\Entity\EntityInterface $entity
+   *   Drupal Entity.
+   *
+   * @return bool
+   *   Is new version.
+   */
+  protected function isNewRevision(EntityInterface $entity) {
+    if ($entity->getEntityTypeId() == "node") {
+      $revision_ids = \Drupal::entityTypeManager()->getStorage($entity->getEntityTypeId())->revisionIds($entity);
+      return count($revision_ids) > 1;
+    }
+    elseif ($entity->getEntityTypeId() == "media") {
+      $mediaStorage = \Drupal::entityTypeManager()->getStorage($entity->getEntityTypeId());
+      return count($this->getRevisionIds($entity, $mediaStorage)) > 1;
+    }
+  }
+
+  /**
+   * Method to get the revisionIds of a media object.
+   *
+   * @param \Drupal\media\Entity\Media $media
+   *   Media object.
+   * @param \Drupal\Core\Entity\EntityStorageInterface $media_storage
+   *   Media Storage.
+   */
+  protected function getRevisionIds(Media $media, EntityStorageInterface $media_storage) {
+    $result = $media_storage->getQuery()
+      ->allRevisions()
+      ->condition($media->getEntityType()->getKey('id'), $media->id())
+      ->sort($media->getEntityType()->getKey('revision'), 'DESC')
+      ->pager(50)
```

Comment:
While using pager is fine with how we use the function now (we only check if there is more than one); if someone re-uses the function call and does need all of the revision ids they will only get the most recent 50 resulting, potentially, in a hidden logic error for media with 50+ revisions.

Reply:
should be fixed with 0f91ce5